packages/core/src/App/Containers/RealAccountSignup/address-details.jsx

```diff
@@ -83,11 +83,21 @@ class AddressDetails extends React.Component {
                 validateOnMount
                 onSubmit={(values, actions) => {
                     if (values.address_state) {
-                        values.address_state = this.props.states_list.length
-                            ? this.state.address_state_to_display
-                                ? getLocation(this.props.states_list, this.state.address_state_to_display, 'value')
-                                : getLocation(this.props.states_list, values.address_state, 'value')
-                            : values.address_state;
+                        let add_state;
+                        if (this.props.states_list.length) {
+                            if (this.state.address_state_to_display) {
+                                add_state = getLocation(
+                                    this.props.states_list,
+                                    this.state.address_state_to_display,
+                                    'value'
+                                );
+                            } else {
+                                add_state = getLocation(this.props.states_list, values.address_state, 'value');
+                            }
+                        } else {
+                            add_state = values.address_state;
+                        }
+                        values.address_state = add_state;
```

Comment:
am just afraid if values is immutable, if its not then its fine 👍 , can try check https://formik.org/docs/api/formik#setfieldvalue-field-string-value-any-shouldvalidate-boolean--void

Reply:
Code has been updated also values is not immutable. tested it out and it is not throwing any error.